python/pyspark/tests/test_util.py

```diff
@@ -73,6 +75,15 @@ def test_parsing_version_string(self):
 
         self.assertRaises(ValueError, lambda: VersionUtils.majorMinorVersion("abced"))
 
+    def test_find_spark_home(self):
+        # SPARK-38827: Test find_spark_home without `SPARK_HOME` environment variable set.
+        origin = os.environ["SPARK_HOME"]
```

Comment:
Note: If `_find_spark_home` doesn't work for some reason, will raise in here directly with `KeyError` to warn users about regression. I think it's reasonable